### ISSUE TITLE
Thinking: Refactoring consul/base.py

### ISSUE THREAD

**Issue by cruatta** [cablehead/python-consul#68]
Any thoughts about refactoring the classes in base.py it into separate modules? Is this something you'd like to see @cablehead?

**Comment by cablehead** [comment]
Yeah, it's not too unruly for me, but it is on the edge. We're at 1740 lines now. You're thinking split it by top level endpoint? kv, event, etc. Pros would it'd easier to focus on a given endpoint. Con would be it'd be harder to do an API sweep to check things are consistent.

**Comment by cablehead** [comment]
I think the time has come to make this split.

I'm thinking:

0. make the spilt by top level endpoint name
0. think through if there's something we could do with broadly common parameters to have them applied consistently, and succinctly, so you can see at a glance they've been applied to an endpoint. challenge for this one is to do it in a way that isn't verbose, and doesn't involve magic.

BTW @cruatta you've been one of the best contributors for this project. Would you be interested in having admin access to be able to help maintain it going forward? @abn is also a maintainer now and it's been great to have the project keep moving forward when I'm unavailable.

**Comment by cablehead** [comment]
An example of where it's currently easy for endpoints to not get consistent treatment: https://github.com/cablehead/python-consul/issues/116

**Comment by cruatta** [comment]
Hey thanks Andy! I can't promise that I have a lot of free time but I'd be glad to help in any way I can, including putting some time towards the split.

**Comment by cablehead** [comment]
This is just awful. I hate magic. But just brain storming ways to make it easy to consistently and explicitly apply parameters - while preserving method signature and docstrings for help and autodoc: https://gist.github.com/cablehead/7fc6d015a3a0c58d7ffdf50df9aaed0d
